source/libtomo/accel/gpu/common.cu

```diff
@@ -284,8 +284,15 @@ cuda_device_query()
     else
         printf("Detected %d CUDA capable devices\n", deviceCount);
 
+    int specific_device = GetEnv("TOMOPY_DEVICE_NUM", -1);
```

Comment:
No, this env variable is effectively the way that a GPU number gets assigned in multi-gpu node: each sub-process in the pool gets passed an index and it sets `"TOMOPY_DEVICE_NUM"` via that index. But I put this here because the migration from threads to processes resulted in every process printing out every single GPU, i.e. when I ran at NERSC, I ended up with 64 printouts of 8 GPUs when 8 processes were started.

Reply:
Here is the relevant impl in algorithm.py:

```python
def _run_accel_algorithm(idx, _func, tomo, center, recon, *_args, **_kwargs):
    # set the device to the specified index within the new process
    os.environ["TOMOPY_DEVICE_NUM"] = "{}".format(idx)
    # execute the accelerated algorithm
    _func(tomo, center, recon, *_args, **_kwargs)
    # recon is the only important modified data
    return recon
```

As you can see, even if the user were to specify this env variable, it would be ignored.